### PR TITLE
Install unit test files again

### PR DESCRIPTION
**Pull request by pvuorela**
Before switch to cmake on commit 7471153ce the unit test files were
installed, and with cmake there are some parts of it done.
Reintroduce the installation as these are needed if the build
environment is not, and perhap cannot be, used for testing.

Also cope on the qml unit test for the example plugin not available
as it's built optionally.

### CODE REVIEW

**Review by dobey — CHANGES_REQUESTED** [review]
Please split separate changes (any adjusting of test behavior, though I think this bit can be avoided here, and installation of tests).

If we're going to be installing the tests, is there any particular reason to not install the one QML file used by the test you're introduction extra logic into, here?

**Comment by pvuorela** [comment]
Made the test installation optional.

On the qml example file installation it would feel a strange rule to install it if examples are built OR if tests are built and installed. Then again the whole double role of the qml file is suspicious. Would be cleaner if there was a qml file as an example and another for testing. Since nobody else has seemed to miss installable tests for a while and the test is small, I can live with just skipping it for now if the file is not available.

**Comment by dobey** [comment]
> On the qml example file installation it would feel a strange rule to install it if examples are built OR if tests are built and installed. Then again the whole double role of the qml file is suspicious. Would be cleaner if there was a qml file as an example and another for testing. Since nobody else has seemed to miss installable tests for a while and the test is small, I can live with just skipping it for now if the file is not available.

Well, it's just a QML file, so there's nothing to "build" for examples. So let's just move it into the tests tree instead? Maybe all the examples should just be part of tests instead of being examples, and anything which is problematic we should just get rid of at this point?

**Review by jpetersen — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -345,9 +346,6 @@ if(enable-examples)
                 examples/plugins/cxx/override/overrideplugin.cpp
                 examples/plugins/cxx/override/overrideplugin.h)
     target_link_libraries(cxxoverrideplugin maliit-plugins Qt5::Widgets)
```
Comment (dobey) [inline CMakeLists.txt:348]: I think you need to also remove the `cxxoverrideplugin` from examples here, since you've removed the files in the one commit?

**Review by dobey — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (pvuorela) [reply CMakeLists.txt:348]: I removed unused qml override files, didn't touch the c++ one.

**Comment by pvuorela** [comment]
@dobey anything else here?

**Review by dobey — APPROVED** [review]
Looks OK now.

**Pull request merged**